core/src/main/scala/org/bitcoins/core/util/Bech32.scala

```diff
@@ -110,27 +113,60 @@ sealed abstract class Bech32 {
     payload
   }
 
-  /** Takes a base32 byte array and encodes it to a string */
-  def encodeToString(b: Vector[UInt8]): String = {
+  /**
+   * Converts a base8 byte vector to base5 byte vector
+   * and then serializes to bech32
+   */
+  def encodeBase8ToString(bytes: ByteVector): String = {
+    val vec = UInt8.toUInt8s(bytes)
+    encodeBase5ToString(vec)
+  }
+
+  /**
+   * Converts a base8 byte vector to base5 byte vector
+   * and then serializes to bech32
+   */
+  def encodeBase8ToString(bytes: Vector[UInt8]): String = {
+    val base5 = fromBase8ToBase5(bytes)
+    encodeBase5ToString(base5)
+  }
+
+  /** Takes a bech32 (base 5) byte array and encodes it to a string */
+  def encodeBase5ToString(b: Vector[UInt8]): String = {
+    require(!b.exists(_ >= u8ThirtyTwo), s"Number out of bech32 range in vector $b")
     b.map(b => charset(b.toInt)).mkString
   }
 
   /** Converts a byte vector from base 8 to base 5 */
-  def fromBase8ToBase5(bytes: ByteVector): Try[Vector[UInt8]] = {
+  def fromBase8ToBase5(bytes: ByteVector): Vector[UInt8] = {
```

Comment:
note all of these names are misleding, it isn't really base8. What it really should be named to is something like 

`byteTo5bit`